algosdk/abi/contract.py

```diff
@@ -10,22 +10,32 @@ class Contract:
 
     Args:
         name (string): name of the contract
-        app_id (int): application id associated with the contract
         methods (list): list of Method objects
+        desc (string, optional): description of the contract
+        networks (dict, optional): information about the contract in a particular network,
+            such as app-id.
     """
 
-    def __init__(self, name: str, app_id: int, methods: List[Method]) -> None:
+    def __init__(
+        self,
+        name: str,
+        methods: List[Method],
+        desc: str = None,
+        networks: Dict[str, dict] = None,
```

Comment:
I'd recommend making a new class to hold the network-specific values instead of using a dictionary, since it's more type safe. E.g. `networks: Dict[str, NetworkInfo]` or similar